### PR TITLE
[docs] Replace @mui/styles on demos

### PR DESCRIPTION
**Pull request by m4theushw**
Related to #2991

This PR replaces `@mui/styles` from all demos and uses the `sx` prop or styled components. I'll open other PRs to migrate `x-grid-data-generator` and other parts still using the legacy styling solution.

The differences pointed by argos are actually the correct screenshots now. They were wrong in the past because the custom styles, which were based on jss, were being inserted before the DataGrid styles, based on emotion, so the emotion styles always won. Now, everything is based on emotion so custom styles work correctly.

Before: _(from the quick filter demo)_

image

After: 

image

### CODE REVIEW

File: docs/src/pages/components/data-grid/columns/RenderExpandCellGrid.js [hunk 1]
```diff
@@ -1,29 +1,11 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import Box from '@mui/material/Box';
 import Typography from '@mui/material/Typography';
 import Paper from '@mui/material/Paper';
 import Popper from '@mui/material/Popper';
-import { createStyles, makeStyles } from '@mui/styles';
 import { DataGrid } from '@mui/x-data-grid';
 
-const useStyles = makeStyles(() =>
```
Comment (mnajdova) [inline docs/src/pages/components/data-grid/columns/RenderExpandCellGrid.js:9]: I would have just converted this to a styled component, but this works too. I usually made this decision but the length of the `sx` prop, if it doesn't fit in one line, I would use styled().

File: docs/src/pages/components/data-grid/components/CustomPaginationGrid.js [hunk 2]
```diff
@@ -1,23 +1,15 @@
 import * as React from 'react';
-import { makeStyles } from '@mui/styles';
+import Box from '@mui/material/Box';
 import { DataGrid, useGridApiContext, useGridState } from '@mui/x-data-grid';
 import { useDemoData } from '@mui/x-data-grid-generator';
 import Pagination from '@mui/material/Pagination';
 
-const useStyles = makeStyles({
-  root: {
-    display: 'flex',
```
Comment (mnajdova) [inline docs/src/pages/components/data-grid/components/CustomPaginationGrid.js:9]: Was this not needed?

File: docs/src/pages/components/data-grid/demo/FullFeaturedDemo.js [hunk 3]
```diff
@@ -306,16 +274,33 @@ export default function FullFeaturedDemo() {
     });
   };
 
+  const DataGridComponent = isAntDesign ? AntDesignStyledDataGridPro : DataGridPro;
+
   return (
-    <div className={classes.root}>
+    <Box
+      sx={{
```
Comment (mnajdova) [inline docs/src/pages/components/data-grid/demo/FullFeaturedDemo.js:281]: Again I would use here `styled()` as the `sx` prop value is too long.

File: docs/src/pages/components/data-grid/editing/ConditionalValidationGrid.js [hunk 4]
```diff
@@ -95,14 +71,22 @@ export default function ConditionalValidationGrid() {
...
-    <div style={{ height: 400, width: '100%' }}>
-      <DataGridPro
-        apiRef={apiRef}
-        className={classes.root}
-        rows={rows}
-        columns={columns}
-        editMode="row"
-      />
-    </div>
+    <Box
```
Comment (mnajdova) [inline docs/src/pages/components/data-grid/editing/ConditionalValidationGrid.js:74]: Prefer `styled()`

File: docs/src/pages/components/data-grid/editing/ValidateServerNameGrid.js [hunk 5]
```diff
@@ -70,15 +48,28 @@ export default function ValidateServerNameGrid() {
   }, []);
 
   return (
-    <div style={{ height: 400, width: '100%' }}>
+    <Box
```
Comment (mnajdova) [inline docs/src/pages/components/data-grid/editing/ValidateServerNameGrid.js:51]: Prefer `styled()`

File: docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.tsx [hunk 6]
```diff
@@ -78,8 +56,22 @@ function QuickSearchToolbar(props: QuickSearchToolbarProps) {
             </IconButton>
           ),
         }}
+        sx={{
+          width: {
+            xs: 1,
+            sm: 'auto',
+          },
+          m: (theme) => theme.spacing(1, 0.5, 1.5),
```
Comment (mnajdova) [inline docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.tsx:64]:
```suggestion
          mt: 1,
          mx: 0.5,
          mb: 1.5,
```
One alternative, not sure if it is worthed.

**Review by mnajdova — APPROVED** [review]
Looks good, I left some comments on places where I think `styled()` is more suitable, but it's not necessary to be changed

File: docs/src/pages/components/data-grid/components/CustomPaginationGrid.js (hunk 2 above)
Comment (m4theushw) [reply docs/src/pages/components/data-grid/components/CustomPaginationGrid.js:9]: It does nothing. In other demos we don't use `display: 'flex'`.

**Comment by m4theushw** [comment]
@mnajdova Thanks for the suggestions.

**Pull request merged**